eppo/src/androidTest/java/cloud/eppo/android/EppoClientTest.java

```diff
@@ -258,6 +259,98 @@ public void testErrorGracefulModeOff() {
                 "subject1", "experiment1", new Attributes(), mapper.readTree("{}")));
   }
 
+  private static EppoHttpClient mockHttpError() {
+    // Create a mock instance of EppoHttpClient
+    EppoHttpClient mockHttpClient = mock(EppoHttpClient.class);
+
+    // Mock sync get
+    when(mockHttpClient.get(anyString())).thenThrow(new RuntimeException("Intentional Error"));
+
+    // Mock async get
+    CompletableFuture<byte[]> mockAsyncResponse = new CompletableFuture<>();
+    when(mockHttpClient.getAsync(anyString())).thenReturn(mockAsyncResponse);
+    mockAsyncResponse.completeExceptionally(new RuntimeException("Intentional Error"));
+
+    return mockHttpClient;
+  }
+
+  @Test
+  public void testGracefulInitializationFailure() {
+    // Set up bad HTTP response
+    EppoHttpClient http = mockHttpError();
+    setBaseClientHttpClientOverrideField(http);
+
+    EppoClient.Builder clientBuilder =
+        new EppoClient.Builder(DUMMY_API_KEY, ApplicationProvider.getApplicationContext())
+            .forceReinitialize(true)
+            .isGracefulMode(true);
+
+    // Initialize and no exception should be thrown.
+    try {
+      clientBuilder.buildAndInitAsync().get();
+    } catch (Exception e) {
+      fail("Unexpected exception thrown: " + e.getMessage());
```

Comment:
nit: you don't really need the try-catch, as any exception will fail the test, but not a big deal if you like it

Reply:
👍